SU2_CFD/src/numerics/NEMO/convection/ausm_slau.cpp

```diff
@@ -0,0 +1,638 @@
+  da_R[nSpecies+nDim]   = 1.0/(2.0*Density_j*SoundSpeed_j) * ((1.0+dPdU_j[nSpecies+nDim])*dPdU_j[nSpecies+nDim]);
+
+  /*--- Sound speed derivatives: Vib-el energy ---*/
+  da_L[nSpecies+nDim+1] = 1.0/(2.0*Density_i*SoundSpeed_i) * ((1.0+dPdU_i[nSpecies+nDim])*dPdU_i[nSpecies+nDim+1]);
+  da_R[nSpecies+nDim+1] = 1.0/(2.0*Density_j*SoundSpeed_j) * ((1.0+dPdU_j[nSpecies+nDim])*dPdU_j[nSpecies+nDim+1]);
+
+  /*--- Left state Jacobian ---*/
+  if (M_F >= 0) {
+
+    /*--- Jacobian contribution: dFc terms ---*/
+    for (auto iVar = 0u; iVar < nSpecies+nDim; iVar++) {
+      for (unsigned short jVar = 0; jVar < nVar; jVar++) {
+        Jacobian_i[iVar][jVar] += M_F * Fc_L[iVar] * da_L[jVar];
+      }
+      Jacobian_i[iVar][iVar] += M_F * SoundSpeed_i;
+    }
+    for (unsigned short iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+      Jacobian_i[nSpecies+nDim][iSpecies] += M_F * (dPdU_i[iSpecies]*SoundSpeed_i + Density_i*Enthalpy_i*da_L[iSpecies]);
+    }
+    for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+      Jacobian_i[nSpecies+nDim][nSpecies+iDim] += M_F * (-dPdU_i[nSpecies+nDim]*Velocity_i[iDim]*SoundSpeed_i + Density_i*Enthalpy_i*da_L[nSpecies+iDim]);
+    }
+    Jacobian_i[nSpecies+nDim][nSpecies+nDim]   += M_F * ((1.0+dPdU_i[nSpecies+nDim])*SoundSpeed_i + Density_i*Enthalpy_i*da_L[nSpecies+nDim]);
+    Jacobian_i[nSpecies+nDim][nSpecies+nDim+1] += M_F * (dPdU_i[nSpecies+nDim+1]*SoundSpeed_i + Density_i*Enthalpy_i*da_L[nSpecies+nDim+1]);
+    for (unsigned short jVar = 0; jVar < nVar; jVar++) {
+      Jacobian_i[nSpecies+nDim+1][jVar] +=  M_F * Fc_L[nSpecies+nDim+1] * da_L[jVar];
+    }
+    Jacobian_i[nSpecies+nDim+1][nSpecies+nDim+1] += M_F * SoundSpeed_i;
+  }
+
+  /*--- Calculate derivatives of the split pressure flux ---*/
+  if ((M_F >= 0) || ((M_F < 0) && (fabs(M_F) <= 1.0))) {
+
+    if (fabs(M_L) <= 1.0) {
+
+      /*--- Mach number ---*/
+      for (unsigned short iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+        dM_LP[iSpecies] = 0.5*(M_L+1.0) * (-ProjVelocity_i/(Density_i*SoundSpeed_i) - ProjVelocity_i*da_L[iSpecies]/(SoundSpeed_i*SoundSpeed_i));
+      for (unsigned short iDim = 0; iDim < nDim; iDim++)
+          dM_LP[nSpecies+iDim] = 0.5*(M_L+1.0) * (-ProjVelocity_i/(SoundSpeed_i*SoundSpeed_i) * da_L[nSpecies+iDim] + UnitNormal[iDim]/(Density_i*SoundSpeed_i));
+      dM_LP[nSpecies+nDim]   = 0.5*(M_L+1.0) * (-ProjVelocity_i/(SoundSpeed_i*SoundSpeed_i) * da_L[nSpecies+nDim]);
+      dM_LP[nSpecies+nDim+1] = 0.5*(M_L+1.0) * (-ProjVelocity_i/(SoundSpeed_i*SoundSpeed_i) * da_L[nSpecies+nDim+1]);
+
+      /*--- Pressure ---*/
+      for(unsigned short iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+        dP_LP[iSpecies] = 0.25*(M_L+1.0) * (dPdU_i[iSpecies]*(M_L+1.0)*(2.0-M_L)
+                                         + Pressure_i*(-ProjVelocity_i/(Density_i*SoundSpeed_i)
+                                         - ProjVelocity_i*da_L[iSpecies]/(SoundSpeed_i*SoundSpeed_i))*(3.0-3.0*M_L));
```

Comment:
Quick search and replace, should make the code less verbose.
```suggestion
                                         - ProjVelocity_i*da_L[iSpecies]/pow(SoundSpeed_i, 2))*(3.0-3.0*M_L));
```